ReactiveXComponent/RabbitMq/RabbitMqSubscriber.cs

```diff
@@ -0,0 +1,237 @@
+                CreateExchangeChannel(exchangeName, routingKey, out channel, out subscriber);
+                if (channel == null || subscriber == null) return;
+                var rabbitMqSubscriberInfos = new RabbitMqSubscriberInfos
+                {
+                    Channel = channel,
+                    Subscriber = subscriber
+                };
+
+                var subscriberKey = new SubscriberKey(_xcConfiguration.GetComponentCode(component), _xcConfiguration.GetStateMachineCode(component, stateMachine));
+                _subscribers.AddOrUpdate(subscriberKey, rabbitMqSubscriberInfos, (key, oldValue) => rabbitMqSubscriberInfos);
+                ReceiveMessage(subscriberKey);
+            }
+            catch (OperationInterruptedException e)
+            {
+                throw new XComponentException("Failed to init Subscriber:" + e.Message, e);
+            }
+        }
+
+        private void CreateExchangeChannel(string exchangeName, string routingKey, out IModel channel,
+            out EventingBasicConsumer subscriber)
+        {
+            if (_connection == null || !_connection.IsOpen)
+            {
+                channel = null;
+                subscriber = null;
+                return;
+            }
+            channel = _connection.CreateModel();
+
+            channel.ModelShutdown += ChannelOnModelShutdown;
+            channel.ExchangeDeclare(exchangeName, ExchangeType.Topic);
+            var queueName = channel.QueueDeclare().QueueName;
+            subscriber = new EventingBasicConsumer(channel);
+            channel.BasicConsume(queueName, false, subscriber);
+            channel.QueueBind(queueName, exchangeName, routingKey, null);
+        }
+
+        private void ReceiveMessage(SubscriberKey subscriberKey)
+        {
+            RabbitMqSubscriberInfos rabbitMqSubscriberInfos;
+            _subscribers.TryGetValue(subscriberKey, out rabbitMqSubscriberInfos);
+            if (rabbitMqSubscriberInfos == null) return;
+            try
+            {
+                rabbitMqSubscriberInfos.Subscriber.Received += (o, e) =>
+                {
+                    rabbitMqSubscriberInfos.Channel?.BasicAck(e.DeliveryTag, false);
+                    DispatchMessage(e);
+                };
+            }
+            catch (EndOfStreamException ex)
+            {
+                ConnectionFailed?.Invoke(this, "Subscriber has been interrupted : " + ex.Message);
+            }    
+        }
+
+        private void AddListenerToRepository(SubscriberKey subscriberKey, Action<MessageEventArgs> stateMachineListener)
+        {
+            _listenerBySubscriberKeyRepo.AddOrUpdate(subscriberKey, 
+            key => new ThreadSafeList<Action<MessageEventArgs>> { stateMachineListener },
+            (key, oldValue) =>
+            {
+                oldValue.Add(stateMachineListener);
+                return oldValue;
+            });
+        }
+
+        public event EventHandler<string> ConnectionFailed;
```

Comment:
If this event's use is only internal to the class or library you should change its access to private or internal.